src/engines/preprocessing_engine.py

```diff
@@ -46,16 +47,25 @@ def clean_data(self, op_type):
                 self.cleaner.remove_nulls()
 
     def has_rows_with_nulls(self, columns):
-        return self.state.imported_data[columns].isnull().values.any()
+        return self.state.raw_data[columns].isnull().values.any()
 
     def reduce_dimensions(self, dim_number=None):
-        self.reducer.reduce(dim_number)
+        return self.reducer.reduce(dim_number)
 
     def number_of_numeric_columns(self):
         if self.state.imported_data is not None:
             return len(self.state.imported_data.select_dtypes(include=np.number).columns.to_list())
         return 0
 
-    def rename_column(self, index, newHeader):
+    def rename_column(self, index, new_header):
         data = self.state.imported_data
-        self.state.imported_data = data.rename(columns={data.columns[index]: newHeader})
+        self.state.imported_data = data.rename(columns={data.columns[index]: new_header})
```

Comment:
I think the names of the columns remain unchanged in the `reduced_columns` list which may cause problems? perhaps that's why they still show up unchanged in the "Columns" group widget or maybe that's unrelated?